### PR TITLE
Improve project props management

### PR DESCRIPTION
**Pull request by reyang**
## Changes

* Moved the top level `*.props` file to `./build/` folder.
* Extracted package version from individual project files to a central place (I only set an example instead of changing all of them, clean up work can be a separate PR to keep this PR small).
* Covered more file types during sanity check.

For significant contributions please make sure you have completed the following items:

* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: build/Common.props [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<Project>
+  <PropertyGroup>
+    <LangVersion>8.0</LangVersion>
+  </PropertyGroup>
+
+  <PropertyGroup Label="Package versions used in this repository">
+    <MicrosoftNETFrameworkReferenceAssembliesPkgVer>[1.0.0]</MicrosoftNETFrameworkReferenceAssembliesPkgVer>
```
Comment (reyang) [inline build/Common.props:7]: I think eventually we want all to extract package version from individual projects to this central file.
We probably should have another place (e.g. `Common.test.props`) for test dependency.

**Comment by reyang** [comment]
The integration test docker files have hard-coded logic (e.g. copy proj/config files) which is different than the normal build flow, I'll need to either change it (by updating the hardwired logic) or fundamentally change it (by using the normal build flow, which means we copy more stuff while building the docker image).

@CodeBlanch FYI

**Comment by CodeBlanch** [comment]
@reyang I made the images as small as possible but that isn't necessarily a hard requirement. At the time, it just seemed like the right thing to do. But we're not shipping them, so probably better to optimize for maintainability? Meaning, if you want to make them just "copy *" I'm good with that 😄

File: OpenTelemetry.sln [hunk 2]
```diff
@@ -11,20 +11,20 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution
 	ProjectSection(SolutionItems) = preProject
 		.dockerignore = .dockerignore
 		.editorconfig = .editorconfig
-		CHANGELOG.md = CHANGELOG.md
```
Comment (CodeBlanch) [inline OpenTelemetry.sln:14]: Why remove CHANGELOG.md from the solution?

Comment (reyang) [reply OpenTelemetry.sln:14]: There is no global CHANGELOG.md, each package has its own changelog, which will be used to generate a single/combined release notes during release.

**Pull request merged**